src/combinatorics/simple_graph/basic.lean

```diff
@@ -122,13 +149,63 @@ set.to_finset G.edge_set
   e ∈ G.edge_finset ↔ e ∈ G.edge_set :=
 by { dunfold edge_finset, simp }
 
+@[simp] lemma edge_set_univ_card [decidable_eq V] [fintype V] [decidable_rel G.adj] :
+  (univ : finset G.edge_set).card = G.edge_finset.card :=
+fintype.card_of_subtype G.edge_finset (mem_edge_finset _)
+
 @[simp] lemma irrefl {v : V} : ¬G.adj v v := G.loopless v
 
 @[symm] lemma edge_symm (u v : V) : G.adj u v ↔ G.adj v u := ⟨λ x, G.sym x, λ x, G.sym x⟩
 
 @[simp] lemma mem_neighbor_set (v w : V) : w ∈ G.neighbor_set v ↔ G.adj v w :=
 by tauto
 
+@[simp] lemma mem_incidence_set (v w : V) : ⟦(v, w)⟧ ∈ G.incidence_set v ↔ G.adj v w :=
+by { dsimp [incidence_set], simp }
```

Comment:
Golf: `by simp [incidence_set]`

Reply:
Should it be `by simp only [incidence_set]`? or is just `simp` fine

Reply:
`simp` should be fine, since it's a terminal `simp`.  It's just that `simp` can also unfold definitions like `dsimp`, so you can combine them.